server/data_provider_canvas_esb.rb

```diff
@@ -66,25 +66,80 @@ def initConfigureCanvasESBProvider(config_hash)
-  def canvasESBToDoLMS(uniqname, security_file, esb_application)
+
+  def canvasAPICalendarEventsURL(uniqname, canvas_courses)
+
+    ## calculate a 15 day range around now for the assignment search (last week, today, next week)
+    ## API call would support specifying attributes to ignore if we deem that necessary at some point.
+    ## https://canvas.instructure.com/doc/api/calendar_events.html
+
+    now = DateTime.now
+    start_date = now.prev_day(7)
+    end_date = now.next_day(8)
+
+    logger.debug "#{self.class.to_s}:#{__method__}:#{__LINE__}: start: #{start_date} now: #{now} end: #{end_date}"
+
+    request_url = "/users/self/calendar_events"
+    request_parameters = {:params => {:as_user_id => "sis_login_id:#{uniqname}",
+                                      :type => 'assignment',
+                                      :start_date => start_date,
+                                      :end_date => end_date,
+                                      :page_size => 100
+    }}
+
+    # Generate url with query parameters.
+    string_request_url = RestClient::Request.new(:method => :get, :url => request_url, :headers => request_parameters).url
+    #### test to see if need to unescape :
+    string_request_url.gsub!(/%3A/, ':')
+    #puts string_request_url.inspect
+    full_request_url = string_request_url
+
+    ### add repeating query parameters.  This is specific to course list
+    full_request_url << CourseList.course_list_string(canvas_courses)
+
+    logger.debug "#{self.class.to_s}:#{__method__}:#{__LINE__}: full_request_url: #{full_request_url}"
+    full_request_url
+  end
+
+  #  <canvas server>/api/v1/users/self/calendar_events
+  def canvasESBToDoLMS(uniqname, canvas_courses, security_file, esb_application)
+    canvas_courses ||=  []
     logger.debug "#{self.class.to_s}:#{__method__}:#{__LINE__}: ############### call canvas ESB todolms esb_application: #{esb_application}"
-    logger.debug "##{self.class.to_s}:#{__method__}:#{__LINE__}: canvas ESB: @canvasESB_w: [#{@canvasESB_w}]"
+    logger.debug "#{self.class.to_s}:#{__method__}:#{__LINE__}: canvas ESB: @canvasESB_w: [#{@canvasESB_w}]"
+    logger.debug "#{self.class.to_s}:#{__method__}:#{__LINE__}: canvas_courses count: #{canvas_courses.length} canvas_courses: #{canvas_courses}"
+
+    calendar_events_url = canvasAPICalendarEventsURL(uniqname, canvas_courses)
 
-    r = @canvasESB_w.get_request "/users/self/upcoming_events?as_user_id=sis_login_id:#{uniqname}"
+    r = @canvasESB_w.get_request calendar_events_url
 
     canvas_body = r.result
     canvas_body_ruby = JSON.parse canvas_body
 
-    return WAPIResultWrapper.new(WAPI::SUCCESS, "got todos from canvas esb", canvas_body_ruby)
+    return WAPIResultWrapper.new(WAPI::SUCCESS, "got calendar_events from canvas esb", canvas_body_ruby)
   end
 
+
+  #  <canvas server>/api/v1/users/self/upcoming_events
+  # actually call out to canvas and return the value.  Caller will reformat if necessary.
```

Comment:
looks like we no longer need this function, since canvasESBToDoLMS captures both prior and upcoming events. This can be removed.